BlockEntity/BECharcoalPit.cs

```diff
@@ -203,7 +200,7 @@ void ConvertPit()
                 int maxY = val.Value.Z;
                 while (lpos.Y <= maxY)
                 {
-                    if (BlockFirepit.IsFirewoodPile(Api.World, lpos))  // test for the possibility someone had contiguous firewood both above and below a soil block for example
+                    if (BlockFirepit.IsFirewoodPile(Api.World, lpos) || lpos == Pos)  // test for the possibility someone had contiguous firewood both above and below a soil block for example

```

Comment:
The addition of lpos == Pos here removes the need for the SetBlock(0, Pos) lower down while also making sure that the firepit block doesn’t end up being a pocket of air in the middle of a column of charcoal if it was buried in the middle of the firewood.